inventory_management_system_api/services/utils.py

```diff
@@ -184,22 +184,20 @@ def _check_missing_mandatory_catalogue_item_properties(
             )
 
 
-def _filter_matching_catalogue_item_properties(
-    defined_properties: Dict[str, Dict],
-    supplied_properties: Dict[str, Dict],
+def _merge_non_mandatory_catalogue_item_properties(
+    defined_properties: Dict[str, Dict], supplied_properties: Dict[str, Dict]
 ) -> Dict[str, Dict]:
     """
-    Filter through the supplied properties and extract the ones matching the defined properties.
-
-    :param defined_properties: The defined catalogue item properties stored as part of the catalogue category in the
-        database.
-    :param supplied_properties: The supplied catalogue item properties.
-    :return: The supplied properties that are matching the defined properties.
+    Merges in any non-mandatory properties that have not been supplied, giving them a value of None, using
+    the same order as they are defined.
```

Comment:
Can we have a note here to say that any extra undefined properties that get supplied are ignored?